components/Match/Match.jsx

```diff
@@ -1,10 +1,59 @@
 import React from 'react';
-import { MatchTable } from '../Table';
-import Error from '../Error';
-
-export default ({ params }) => (
-  <div>
-    {params && params.match_id && <MatchTable matchId={params.match_id} />}
-    {(!params || !params.match_id) && <Error />}
-  </div>
-);
+import { Table } from '../Table';
+// import Error from '../Error';
+import { REDUCER_KEY } from '../../reducers';
+import { getMatch } from '../../actions';
+import { connect } from 'react-redux';
+import createConstantsWrapper from '../Constants';
+
+const columns = [{
+  displayName: 'Hero',
+  field: 'hero_id',
+  width: 2,
+  sorted: false,
+}, {
+  displayName: 'Kills',
+  field: 'kills',
+  width: 2,
+  sorted: false,
+}, {
+  displayName: 'Deaths',
+  field: 'deaths',
+  width: 2,
+  sorted: false,
+}, {
+  displayName: 'Assists',
+  field: 'assists',
+  width: 2,
+  sorted: false,
+}];
+
+const mapStateToProps = (state) => {
+  const { error, loading, match } = state[REDUCER_KEY].gotMatch;
+  return {
+    loading,
+    error,
+    data: match,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => ({
+  sort: (column) => dispatch(getMatch(column)),
+  getMatch: (matchId) => dispatch(getMatch(matchId)),
+});
+
+class RequestLayer extends React.Component {
+  componentDidMount() {
+    this.props.getMatch(this.props.routeParams.match_id);
+  }
+
+  render() {
+    return (<div>
+      {<Table {...this.props} data={this.props.data && this.props.data.players ? this.props.data.players : this.props.data} columns={columns} />}
+    </div>);
+  }
+}
```

Comment:
Can we put the div on a separate line like so:

```js
return (
  <div>
    <Component />
  </div>
);
```

Also, instead of doing `data={this.props.data ... this.props.data ... this.props.data}`, we can destructure the data object so you don't need to write it so verbosely (making the line longer and harder to read mostly, also a pain to type).

```js
  const { data } = this.props;
  return <div><Table {...this.props} data={data && data.players ? ...
```